ConsumerApi.Tests.Integration/StepDefinitions/TokensApiStepDefinitions.cs

```diff
@@ -1,62 +1,83 @@
-using System.Net;
-using Backbone.ConsumerApi.Tests.Integration.API;
+﻿using Backbone.BuildingBlocks.SDK.Endpoints.Common.Types;
+using Backbone.ConsumerApi.Sdk;
+using Backbone.ConsumerApi.Sdk.Authentication;
+using Backbone.ConsumerApi.Sdk.Endpoints.Tokens.Types;
+using Backbone.ConsumerApi.Sdk.Endpoints.Tokens.Types.Requests;
+using Backbone.ConsumerApi.Sdk.Endpoints.Tokens.Types.Responses;
 using Backbone.ConsumerApi.Tests.Integration.Configuration;
 using Backbone.ConsumerApi.Tests.Integration.Extensions;
-using Backbone.ConsumerApi.Tests.Integration.Models;
-using Backbone.Crypto.Abstractions;
+using Backbone.Crypto;
+using Backbone.Tooling;
 using Microsoft.Extensions.Options;
 using Newtonsoft.Json;
-using TechTalk.SpecFlow.Assist;
 
 namespace Backbone.ConsumerApi.Tests.Integration.StepDefinitions;
 
 [Binding]
 [Scope(Feature = "POST Token")]
 [Scope(Feature = "GET Token")]
 [Scope(Feature = "GET Tokens")]
-internal class TokensApiStepDefinitions : BaseStepDefinitions
+internal class TokensApiStepDefinitions
 {
-    private readonly TokensApi _tokensApi;
     private string _tokenId;
     private string _peerTokenId;
     private readonly List<CreateTokenResponse> _givenOwnTokens;
     private readonly List<Token> _responseTokens;
-    private HttpResponse<CreateTokenResponse>? _createTokenResponse;
-    private HttpResponse<Token>? _tokenResponse;
-    private HttpResponse<IEnumerable<Token>>? _tokensResponse;
-    private static readonly string TOMORROW_AS_STRING = DateTime.Now.AddDays(1).ToString("yyyy-MM-dd");
-    private static readonly string YESTERDAY_AS_STRING = DateTime.Now.AddDays(-1).ToString("yyyy-MM-dd");
-
-    public TokensApiStepDefinitions(IOptions<HttpConfiguration> httpConfiguration, TokensApi tokensApi, ISignatureHelper signatureHelper, ChallengesApi challengesApi, IdentitiesApi identitiesApi, DevicesApi devicesApi) :
-        base(httpConfiguration, signatureHelper, challengesApi, identitiesApi, devicesApi)
+    private ApiResponse<CreateTokenResponse>? _createTokenResponse;
+    private ApiResponse<EmptyResponse>? _createTokenResponseError;
```

Comment:
Why do we need a separate response in case of an error?

Reply:
When met with a `401 Unauthorized`, the response received does not match a typical API Response, thus the Deserialization fails.
For this reason, we an `EmptyResponse`.

Reply:
Damn... we really need a correct response in that case. :( 

What bothers me about your solution is that there's no special response in case of any other error. So it's really confusing to have this special type there.
Therefore I would suggest a `_createTokenResponse401` as a temporary workaround. In the long run we should really return our json error though.